src/openms_gui/source/VISUAL/DIALOGS/TheoreticalSpectrumGenerationDialog.cpp

```diff
@@ -36,29 +36,94 @@
 #include <OpenMS/VISUAL/DIALOGS/TheoreticalSpectrumGenerationDialog.h>
 #include <ui_TheoreticalSpectrumGenerationDialog.h>
 
+#include <OpenMS/CHEMISTRY/TheoreticalSpectrumGenerator.h>
+#include <OpenMS/CHEMISTRY/NucleicAcidSpectrumGenerator.h>
 #include <OpenMS/DATASTRUCTURES/String.h>
 
+// Qt includes
+#include <QtWidgets/QMessageBox>
+#include <qflags.h>
+
 
 namespace OpenMS
 {
+  // Note: If an additional checkbox is added all of the following three objects have to be edited!
+  //
+  //
+  // enum to get ion checkbox index (Ordering has to be the same as in the ui!)
+  enum class Checkbox
+  {
+    A_Ions,
+    A_b_Ions,
+    B_Ions,
+    C_Ions,
+    D_Ions,
+    W_Ions,
+    X_Ions,
+    Y_Ions,
+    Z_Ions,
+    Precursor,
+    Neutral_losses,
+    Abundant_Immonium_Ions
+  };
+
+  // vector of all enum entries for iteration
+  const std::vector<Checkbox> check_box_names {Checkbox::A_Ions, Checkbox::A_b_Ions, Checkbox::B_Ions, Checkbox::C_Ions,    Checkbox::D_Ions,         Checkbox::W_Ions,
+                                               Checkbox::X_Ions, Checkbox::Y_Ions,   Checkbox::Z_Ions, Checkbox::Precursor, Checkbox::Neutral_losses, Checkbox::Abundant_Immonium_Ions};
+
+  // map from checkbox (index) to corresponding parameter with description
+  const std::map<Checkbox, std::pair<String, String>> checkbox_to_param {
+    {Checkbox::A_Ions, {"add_a_ions", "Add peaks of a-ions to the spectrum"}},
+    {Checkbox::A_b_Ions, {"add_a-B_ions", "Add peaks of a-B-ions to the spectrum (nucleotide sequences only)"}},
+    {Checkbox::B_Ions, {"add_b_ions", "Add peaks of b-ions to the spectrum"}},
+    {Checkbox::C_Ions, {"add_c_ions", "Add peaks of c-ions to the spectrum"}},
+    {Checkbox::D_Ions, {"add_d_ions", "Add peaks of d-ions to the spectrum"}},
+    {Checkbox::W_Ions, {"add_w_ions", "Add peaks of w-ions to the spectrum"}},
```

Comment:
Do they work for AAs?

Reply:
not yet, `TheoreticalSpectrumGenerator` doesn't support them

Reply:
just because it does not say (NA only) like the a-B ion

Reply:
ahh, yes I can add that to the description - I just copied the parameter names and descriptions from the `NucleicAcidSpectrumGenerator` constructor

Reply:
Should we maybe hide or delete them when not applicable? Especially if we do Metabolites too.

Reply:
They are disabled and enabled depending on sequence type.
See line 111.

Reply:
Yes but disabling is cluttered. Can we hide?

Reply:
Ahh that's what you mean. Yes, that is possible. I even tried that already but decided against it because I found it distracting when the dialog completely changed its appearance. Disabling just looked smoother for me.

Reply:
if we add glycan or cross-links a fixed list could get out of hands :P
If you have the code still around for removing, I would actually also prefer that. And I guess it is not that distracting because a user will typically stick with one type (e.g., only switch once)

Reply:
Alright, going to change to hiding instead of disabling.